File: source/android/uitestapp/src/main/java/io/adaptivecards/uitestapp/ui/test_cases/TestCasesViewModel.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package io.adaptivecards.uitestapp.ui.test_cases;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class TestCasesViewModel extends ViewModel
{
    private MutableLiveData<String> mLastClickedItem;

    public TestCasesViewModel()
    {
        mLastClickedItem = new MutableLiveData<>();
    }

    public void setLastClickedItem(String s)
    {
        mLastClickedItem.setValue(s);
    }

    public LiveData<String> getLastClickedItem()
    {
        return mLastClickedItem;
    }
}

File: source/android/uitestapp/src/main/java/io/adaptivecards/uitestapp/ui/test_cases/TestCasesViewModel.kt
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package io.adaptivecards.uitestapp.ui.test_cases

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class TestCasesViewModel : ViewModel() {
    private val mLastClickedItem: MutableLiveData<String>

    fun setLastClickedItem(s: String) {
        mLastClickedItem.value = s
    }

    val lastClickedItem: LiveData<String>
        get() = mLastClickedItem

    init {
        mLastClickedItem = MutableLiveData()
    }
}
